tdp/core/collections.py

```diff
@@ -18,8 +21,13 @@
 
 
 class Collections(Mapping):
+    """A mapping of collection name to Collection instance.
+
+    This class also gather operations from all collections and filter them by their presence or not in the DAG.
+    """
+
     def __init__(self, collections):
-        self._collections = collections
+        self._collections: "Collections" = collections
```

Comment:
Why `Collections` is a double quote string ?

Reply:
Because `Collections` hasn't been defined yet, hence it will throw an error at runtime if I don't forward the ref (see  [PEP 484](https://peps.python.org/pep-0484/#forward-references)).

Reply:
Wait why a `Collections` object have a nested `Collections` object ?

Reply:
Actually I dont even know why I put the type here. It should be on the argument, not on the assignation.

And you're right, the correct type is `OrderedDict[string, Collection]`, not `"Collections"`!

Reply:
Done